lib/omnibus/packagers/msi.rb

```diff
@@ -47,24 +47,46 @@ class Packager::MSI < Packager::Base
       FileSyncer.glob("#{resources_path}/assets/*").each do |file|
         copy_file(file, "#{resources_dir}/assets/#{File.basename(file)}")
       end
+
+      # If fastmsi, copy externally built custom action DLL into the staging directory
+      copy_file("#{resources_path}/ext/CustomActionFastMsi.CA.dll", "#{staging_dir}") if fast_msi
     end
 
     build do
+      # If fastmsi, zip up the contents of the install directory
+      zip_command = <<-EOH.split.join(' ').squeeze(' ').strip
```

Comment:
Similar to my comment on heat_command, I would consider making zip_command a method that returns this string.